driver-core/src/main/java/com/datastax/driver/core/BoundStatement.java

```diff
@@ -1104,6 +1117,199 @@ public BoundStatement setPartitionKeyToken(Token v) {
+        return wrapper.getDouble(name);
+    }
+
+    @Override
+    public ByteBuffer getBytesUnsafe(int i) {
+        return wrapper.getBytesUnsafe(i);
+    }
+
+    @Override
+    public ByteBuffer getBytesUnsafe(String name) {
+        return wrapper.getBytesUnsafe(name);
+    }
+
+    @Override
+    public ByteBuffer getBytes(int i) {
+        return wrapper.getBytes(i);
+    }
+
+    @Override
+    public ByteBuffer getBytes(String name) {
+        return wrapper.getBytes(name);
+    }
+
+    @Override
+    public String getString(int i) {
+        return wrapper.getString(i);
+    }
+
+    @Override
+    public String getString(String name) {
+        return wrapper.getString(name);
+    }
+
+    @Override
+    public BigInteger getVarint(int i) {
+        return wrapper.getVarint(i);
+    }
+
+    @Override
+    public BigInteger getVarint(String name) {
+        return wrapper.getVarint(name);
+    }
+
+    @Override
+    public BigDecimal getDecimal(int i) {
+        return wrapper.getDecimal(i);
+    }
+
+    @Override
+    public BigDecimal getDecimal(String name) {
+        return wrapper.getDecimal(name);
+    }
+
+    @Override
+    public UUID getUUID(int i) {
+        return wrapper.getUUID(i);
+    }
+
+    @Override
+    public UUID getUUID(String name) {
+        return wrapper.getUUID(name);
+    }
+
+    @Override
+    public InetAddress getInet(int i) {
+        return wrapper.getInet(i);
+    }
+
+    @Override
+    public InetAddress getInet(String name) {
+        return wrapper.getInet(name);
+    }
+
+    @Override
+    public <T> List<T> getList(int i, Class<T> elementsClass) {
+        return wrapper.getList(i, elementsClass);
+    }
+
+    @Override
+    public <T> List<T> getList(String name, Class<T> elementsClass) {
+        return wrapper.getList(name, elementsClass);
+    }
+
+    @Override
+    public <T> Set<T> getSet(int i, Class<T> elementsClass) {
+        return wrapper.getSet(i, elementsClass);
+    }
+
+    @Override
+    public <T> Set<T> getSet(String name, Class<T> elementsClass) {
+        return wrapper.getSet(name, elementsClass);
+    }
+
+    @Override
+    public <K, V> Map<K, V> getMap(int i, Class<K> keysClass, Class<V> valuesClass) {
+        return wrapper.getMap(i, keysClass, valuesClass);
+    }
+
+    @Override
+    public <K, V> Map<K, V> getMap(String name, Class<K> keysClass, Class<V> valuesClass) {
+        return wrapper.getMap(name, keysClass, valuesClass);
+    }
+
+    /**
+     * Returns all the values bound for this statement.
+     * <p>
+     * Note that this method may have a non-negligible impact on performance: internally, values
+     * are stored in serialized form, so they need to be deserialized one by one. This method is
+     * provided for debugging purposes.
+     *
+     * @return the values, in the order in which they appear in the statement. Any unset value
+     * will be returned as {@code null}.
+     */
+    public List<Object> getValues() {
```

Comment:
Yes, both very good points. I need to check how names work when the query uses positional parameters, but IIRC they get generated.